src/Component/Provider/ArrayGridProvider.php

```diff
@@ -13,21 +13,28 @@
 
 namespace Sylius\Component\Grid\Provider;
 
+use Sylius\Component\Grid\Configuration\GridConfigurationExtenderInterface;
 use Sylius\Component\Grid\Definition\ArrayToDefinitionConverterInterface;
 use Sylius\Component\Grid\Definition\Grid;
 use Sylius\Component\Grid\Exception\UndefinedGridException;
+use Webmozart\Assert\Assert;
 
 final class ArrayGridProvider implements GridProviderInterface
 {
     private ArrayToDefinitionConverterInterface $converter;
+    private GridConfigurationExtenderInterface $gridConfigurationExtender;
 
     /** @var array[] */
     private array $gridConfigurations;
 
-    public function __construct(ArrayToDefinitionConverterInterface $converter, array $gridConfigurations)
-    {
+    public function __construct(
+        ArrayToDefinitionConverterInterface $converter,
+        array $gridConfigurations,
+        GridConfigurationExtenderInterface $gridConfigurationExtender
```

Comment:
This argument should be nullable to maintain BC

Reply:
done